From 73ab55c657adcd5dd5882110ef43fdb833e5f518 Mon Sep 17 00:00:00 2001
From: Arne Coucheron <arco68@gmail.com>
Date: Sun, 9 Oct 2016 06:18:47 +0200
Subject: [PATCH 06/25] stagefright: omx: Don't signal dataspace change on
 legacy QCOM

This isn't supported in legacy media HAL, and causes things like
screen recording and wifi display to fail when setting up the encoder.

Change-Id: Icb3f7b7dfefcfd72939037241568f28c01fc11ed
---
 media/libstagefright/bqhelper/GraphicBufferSource.cpp | 2 ++
 media/libstagefright/omx/Android.bp                   | 8 ++++++++
 2 files changed, 10 insertions(+)

diff --git a/media/libstagefright/bqhelper/GraphicBufferSource.cpp b/media/libstagefright/bqhelper/GraphicBufferSource.cpp
index dd03d3837..e4e9aadb5 100644
--- a/media/libstagefright/bqhelper/GraphicBufferSource.cpp
+++ b/media/libstagefright/bqhelper/GraphicBufferSource.cpp
@@ -816,11 +816,13 @@ status_t GraphicBufferSource::submitBuffer_l(const VideoBuffer &item) {
         return UNKNOWN_ERROR;
     }
 
+#ifndef QCOM_BSP_LEGACY
     if ((android_dataspace)item.mDataspace != mLastDataspace) {
         onDataspaceChanged_l(
                 item.mDataspace,
                 (android_pixel_format)item.mBuffer->getGraphicBuffer()->format);
     }
+#endif
 
     std::shared_ptr<AcquiredBuffer> buffer = item.mBuffer;
     // use a GraphicBuffer for now as component is using GraphicBuffers to hold references
diff --git a/media/libstagefright/omx/Android.bp b/media/libstagefright/omx/Android.bp
index 3e6942bc8..139f37919 100644
--- a/media/libstagefright/omx/Android.bp
+++ b/media/libstagefright/omx/Android.bp
@@ -72,6 +72,14 @@ cc_library_shared {
         "-Wno-documentation",
     ],
 
+    product_variables: {
+        lineage: {
+            uses_qcom_bsp_legacy: {
+                cppflags: ["-DQCOM_BSP_LEGACY"],
+            },
+        },
+    },
+
     sanitize: {
         misc_undefined: [
             "signed-integer-overflow",
-- 
2.17.1

